### PR TITLE
Add Media Device Path subtypes

### PR DESCRIPTION
**Pull request by joschock**
Add sub-type definitions for Media DevicePath type per section 10.3.5 of UEFI Spec 2.10.

### CODE REVIEW

File: src/protocols/device_path.rs [hunk 1]
```diff
@@ -64,3 +64,19 @@ pub struct HardDriveMedia {
     pub partition_format: u8,
     pub signature_type: u8,
 }
+
+pub struct Media {
+    pub header: Protocol,
+}
+
+impl Media {
+    pub const SUBTYPE_HARD_DRIVE: u8 = 0x01;
```
Comment (dvdhrm) [inline src/protocols/device_path.rs:73]: EDK2 uses `MEDIA_HARDDRIVE_DP`, so should be use `SUBTYPE_HARDDRIVE` as well? The spec has no clear identifier for this, but EDK2 uses `HARDDRIVE` consistently throughout the code-base.

File: src/protocols/device_path.rs [hunk 2]
```diff
@@ -64,3 +64,19 @@ pub struct HardDriveMedia {
...
+    pub const SUBTYPE_CD_ROM: u8 = 0x02;
```
Comment (dvdhrm) [inline src/protocols/device_path.rs:74]: Similar thing here, EDK2 uses `CDROM` as single word. `CD_ROM` is not used once.

File: src/protocols/device_path.rs [hunk 3]
```diff
@@ -64,3 +64,19 @@ pub struct HardDriveMedia {
...
+    pub const SUBTYPE_VENDOR: u8 = 0x03;
+    pub const SUBTYPE_FILE_PATH: u8 = 0x04;
+    pub const SUBTYPE_MEDIA_PROTOCOL: u8 = 0x05;
+    pub const SUBTYPE_PIWG_FIRMWARE_FILE: u8 = 0x06;
+    pub const SUBTYPE_PIWG_FIRMWARE_VOLUME: u8 = 0x07;
+    pub const SUBTYPE_RELATIVE_OFFSET_RANGE: u8 = 0x08;
+    pub const SUBTYPE_RAMDISK: u8 = 0x09;
```
Comment (dvdhrm) [inline src/protocols/device_path.rs:81]: Opposite case here, EDK2 uses `RAM_DISK`. There are some other definitions that use `RAMDISK`, but the majority uses `RAM_DISK`. I think we should follow this style.

**Review by dvdhrm — CHANGES_REQUESTED** [review]
Thanks a lot! The spec has no clear identifier names for the media types, so my preference would be to follow what EDK2 uses. See my inline comments for suggestions.

Let me know if I missed something!

**Comment by dvdhrm** [comment]
(I can also amend these suggestions when applying the patches, if you prefer)

File: src/protocols/device_path.rs (hunk 1 above)
Comment (joschock) [reply src/protocols/device_path.rs]: Amended as requested in latest push.

File: src/protocols/device_path.rs (hunk 2 above)
Comment (joschock) [reply src/protocols/device_path.rs]: Amended as requested in latest push.

File: src/protocols/device_path.rs (hunk 3 above)
Comment (joschock) [reply src/protocols/device_path.rs]: Amended as requested in latest push.

**Comment by joschock** [comment]
> (I can also amend these suggestions when applying the patches, if you prefer)

Thanks for the feedback! Appreciate the quick turnaround on patches - I do not have strong feelings on the naming, so was happy to push an update with the requested modifications.

**Comment by dvdhrm** [comment]
Thanks for the update! Merged and pushed!

**Pull request closed without merging**